src/main/scala/com/mongodb/spark/rdd/partitioner/MongoFixedNumberPartitioner.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.spark.rdd.partitioner
+
+import scala.util.{Failure, Success, Try}
+
+import org.bson.{BsonDocument, BsonValue}
+import com.mongodb.MongoCommandException
+import com.mongodb.client.MongoCollection
+import com.mongodb.client.model.{Projections, Sorts}
+import com.mongodb.spark.MongoConnector
+import com.mongodb.spark.config.ReadConfig
+
+/**
+ * The Fixed Number Partitioner.
+ *
+ * Partitions the collection into a fixed number of partitions.
+ *
+ * $configurationProperties
+ *
+ *  - [[partitionKeyProperty partitionkey]], the field to partition the collection by. The field should be indexed and contain unique values.
+ *     Defaults to `_id`.
+ *  - [[numberOfPartitionsProperty numberofpartitions]], the maximum number of partitions to create. Defaults to `64`.
+ *
+ *
+ * '''Note:''' This can be a expensive operation as it creates 1 cursor for every partition.
+ *
+ * @since 1.0
+ */
+class MongoFixedNumberPartitioner extends MongoPartitioner {
```

Comment:
I'm reticent to abstract the common code - as they are nice concrete examples of connectors.

Like the names.